utils/api.js

```diff
@@ -397,6 +400,65 @@ const api = {
     );
   },
 
+  /**
+   * @param {string} groupId
+   * @param {string} fragmentEntryKey
+   * @param {string} thumbnailPath
+   * @param {string} previewFileEntryId
+   */
+  uploadThumbnail(
+    groupId,
+    fragmentEntryKey,
+    thumbnailPath,
+    previewFileEntryId = '0'
+  ) {
+    const bytes = JSON.stringify([...fs.readFileSync(thumbnailPath)]);
+
+    const filename = `${groupId}_${fragmentEntryKey}_${path.basename(
+      thumbnailPath
+    )}`;
+
+    let fileEntry;
+
+    if (Number(previewFileEntryId) > 0) {
+      fileEntry = this.postFormData(
+        '/api/jsonws/dlapp/update-file-entry',
```

Comment:
Are these images visible in Documents and Media?

Reply:
Yep, there is no way to hide it. And we have no D&M API to upload hidden files.